src/apprunner/wizards/apprunnerCreateServiceWizard.ts

```diff
@@ -0,0 +1,227 @@
+        const dontShow = localize('aws.generic.doNotShowAgain', "Don't Show Again")
+        const pricingUri = vscode.Uri.parse('https://aws.amazon.com/apprunner/pricing/')
+
+        vscode.window.showInformationMessage(notice, viewPricing, dontShow).then(button => {
+            if (button === viewPricing) {
+                vscode.env.openExternal(pricingUri)
+                showDeploymentCostNotifcation()
+            } else if (button === dontShow) {
+                ext.context.globalState.update('apprunner.deployments.notifyPricing', false)
+            }
+        })
+    }
+}
+
+function createSourcePrompter(
+    autoDeployButton: QuickInputToggleButton
+): Prompter<CreateServiceRequest['SourceConfiguration']> {
+    const ecrPath = {
+        label: 'ECR',
+        data: { ImageRepository: {} } as ImageRepositorySource,
+        detail: localize(
+            'AWS.apprunner.createService.ecr.detail',
+            'Create a service from a public or private Elastic Container Registry repository'
+        ),
+    }
+
+    const repositoryPath = {
+        label: 'Repository',
+        data: { CodeRepository: {} } as CodeRepositorySource,
+        detail: localize('AWS.apprunner.createService.repository.detail', 'Create a service from a GitHub repository'),
+    }
+
+    return picker.createQuickPick([ecrPath, repositoryPath], {
+        title: localize('AWS.apprunner.createService.sourceType.title', 'Select a source code location type'),
+        buttons: [autoDeployButton, ...makeButtons()],
+    })
+}
+
+export type CreateServiceRequest = Omit<AppRunner.CreateServiceRequest, 'SourceConfiguration'> & {
+    SourceConfiguration: CodeRepositorySource | ImageRepositorySource
+}
+
+export class CreateAppRunnerServiceForm extends WizardForm<CreateServiceRequest> {
+    public constructor(context: CreateAppRunnerServiceContext) {
+        super()
+        const form = this.body
+
+        form.SourceConfiguration.bindPrompter(() => createSourcePrompter(context.autoDeployButton))
+
+        form.SourceConfiguration.applyForm(context.imageRepositoryForm, {
+            showWhen: state => state.SourceConfiguration?.ImageRepository !== undefined,
+        })
+        form.SourceConfiguration.applyForm(context.codeRepositoryForm, {
+            showWhen: state => state.SourceConfiguration?.CodeRepository !== undefined,
+        })
+
+        form.ServiceName.bindPrompter(() =>
+            input.createInputBox({
+                title: localize('AWS.apprunner.createService.name.title', 'Name your service'),
+                validateInput: validateName, // TODO: we can check if names match any already made services
+                buttons: makeButtons(),
+            })
+        )
+
+        form.InstanceConfiguration.bindPrompter(() => createInstanceStep())
+    }
+}
+
+export class CreateAppRunnerServiceWizard extends Wizard<CreateServiceRequest> {
```

Comment:
This is another hint (to me) of what we gain by avoiding extra concepts: if we conflate Wizard+WizardForm, then we avoid having to define wizards _and_ forms for every wizard we want to make. 

Is the distinction between "wizard" and "form" _semantically_ meaningful, or is it an accident of code organization? We should try hard to avoid forcing us to utter meaningless distinctions over and over in code. The names `CreateAppRunnerServiceWizard` and `CreateAppRunnerServiceForm ` aren't really needed by anyone, they don't capture a useful distinction, only a mechanical accident.

To create a Wizard, I want to define one Wizard class. Not a Form, Wizard, Context, etc. Some of these things might be technically unavoidable but any of them that we can avoid will have cascading benefits.

Reply:
Yeah this is mostly a case of organization that _seemed_ reasonable but really isn't that helpful. Within the core of the framework, the decoupling between a Wizard and its 'form' is very important. But for anyone using the framework, that matters much less and only serves to clutter the vocabulary (like you said). Wizards already expose a form field anyway, so there really is not a need to separately instantiate them. 

We can get rid of the 'context' object too. I was mainly following a similar structure to previous wizards, but since this is an entirely new thing we might as well just cut out all the fluff.

Reply:
> To create a Wizard, I want to define one Wizard class.

Just to clarify this: a "one Wizard (potentially containing other Wizards)".

Reply:
> Just to clarify this: "one Wizard (potentially containing other Wizards)".

Agreed. Having just a single `Wizard` makes things very simple.